File: Nx3CS/bin/test/src/nx3/PAttributesRectsCalculator.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace nx3{
	public  class PAttributesRectsCalculator : global::haxe.lang.HxObject {
		public    PAttributesRectsCalculator(global::haxe.lang.EmptyObject empty){
			unchecked {
				#line 8 "F:\\nx3\\src\\nx3\\PAttributesRectsCalculator.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public    PAttributesRectsCalculator(){
			unchecked {
				#line 8 "F:\\nx3\\src\\nx3\\PAttributesRectsCalculator.hx"
				global::nx3.PAttributesRectsCalculator.__hx_ctor_nx3_PAttributesRectsCalculator(this);
			}
			#line default
		}
		
		
		public static   void __hx_ctor_nx3_PAttributesRectsCalculator(global::nx3.PAttributesRectsCalculator __temp_me74){
			unchecked {
				#line 8 "F:\\nx3\\src\\nx3\\PAttributesRectsCalculator.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public static   global::nx3.geom.Rectangle getClefRect(global::nx3.EClef clef){
			unchecked {
				#line 12 "F:\\nx3\\src\\nx3\\PAttributesRectsCalculator.hx"
				switch (global::Type.enumIndex(clef)){
					case 2:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PAttributesRectsCalculator.hx"
						return new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(((double) (0) ), true), new global::haxe.lang.Null<double>(((double) (-3) ), true), new global::haxe.lang.Null<double>(((double) (9) ), true), new global::haxe.lang.Null<double>(((double) (6) ), true));
					}
					
					
					case 1:
					{
						#line 15 "F:\\nx3\\src\\nx3\\PAttributesRectsCalculator.hx"
						return new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(((double) (0) ), true), new global::haxe.lang.Null<double>(((double) (-4) ), true), new global::haxe.lang.Null<double>(((double) (9) ), true), new global::haxe.lang.Null<double>(((double) (8) ), true));
					}
					
					
					case 0:
					{
						#line 16 "F:\\nx3\\src\\nx3\\PAttributesRectsCalculator.hx"
						return new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(((double) (0) ), true), new global::haxe.lang.Null<double>(((double) (-5) ), true), new global::haxe.lang.Null<double>(((double) (9) ), true), new global::haxe.lang.Null<double>(((double) (10) ), true));
					}
					
					
				}
				
				#line 10 "F:\\nx3\\src\\nx3\\PAttributesRectsCalculator.hx"
				return default(global::nx3.geom.Rectangle);
			}
			#line default
		}
		
		
		public static   global::nx3.geom.Rectangle getKeyRect(global::nx3.EKey key){
			unchecked {
				#line 24 "F:\\nx3\\src\\nx3\\PAttributesRectsCalculator.hx"
				switch (global::Type.enumIndex(key)){
					case 6:
					{
						#line 26 "F:\\nx3\\src\\nx3\\PAttributesRectsCalculator.hx"
						return new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(((double) (0) ), true), new global::haxe.lang.Null<double>(((double) (-3) ), true), new global::haxe.lang.Null<double>(((double) (1) ), true), new global::haxe.lang.Null<double>(((double) (6) ), true));
					}
					
					
					case 7:case 5:
					{
						#line 27 "F:\\nx3\\src\\nx3\\PAttributesRectsCalculator.hx"
						return new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(((double) (0) ), true), new global::haxe.lang.Null<double>(((double) (-3) ), true), new global::haxe.lang.Null<double>(4.4, true), new global::haxe.lang.Null<double>(((double) (6) ), true));
					}
					
					
					case 8:case 4:
					{
						#line 28 "F:\\nx3\\src\\nx3\\PAttributesRectsCalculator.hx"
						return new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(((double) (0) ), true), new global::haxe.lang.Null<double>(((double) (-3) ), true), new global::haxe.lang.Null<double>(6.8, true), new global::haxe.lang.Null<double>(((double) (6) ), true));
					}
					
					
					case 9:case 3:
					{
						#line 29 "F:\\nx3\\src\\nx3\\PAttributesRectsCalculator.hx"
						return new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(((double) (0) ), true), new global::haxe.lang.Null<double>(((double) (-3) ), true), new global::haxe.lang.Null<double>(9.2, true), new global::haxe.lang.Null<double>(((double) (6) ), true));
					}
					
					
					case 10:case 2:
					{
						#line 30 "F:\\nx3\\src\\nx3\\PAttributesRectsCalculator.hx"
						return new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(((double) (0) ), true), new global::haxe.lang.Null<double>(((double) (-3) ), true), new global::haxe.lang.Null<double>(11.6, true), new global::haxe.lang.Null<double>(((double) (6) ), true));
					}
					
					
					case 11:case 1:
					{
						#line 31 "F:\\nx3\\src\\nx3\\PAttributesRectsCalculator.hx"
						return new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(((double) (0) ), true), new global::haxe.lang.Null<double>(((double) (-3) ), true), new global::haxe.lang.Null<double>(14.0, true), new global::haxe.lang.Null<double>(((double) (6) ), true));
					}
					
					
					default:
					{
						#line 33 "F:\\nx3\\src\\nx3\\PAttributesRectsCalculator.hx"
						return new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(((double) (0) ), true), new global::haxe.lang.Null<double>(((double) (-2) ), true), new global::haxe.lang.Null<double>(.5, true), new global::haxe.lang.Null<double>(((double) (4) ), true));
					}
					
				}
				
			}
			#line default
		}
		
		
		public static   global::nx3.geom.Rectangle getTimeRect(global::nx3.ETime time){
			unchecked {
				#line 39 "F:\\nx3\\src\\nx3\\PAttributesRectsCalculator.hx"
				if (( time == default(global::nx3.ETime) )) {
					#line 39 "F:\\nx3\\src\\nx3\\PAttributesRectsCalculator.hx"
					return new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(((double) (0) ), true), new global::haxe.lang.Null<double>(((double) (-3) ), true), new global::haxe.lang.Null<double>(.5, true), new global::haxe.lang.Null<double>(((double) (3) ), true));
				}
				
				#line 40 "F:\\nx3\\src\\nx3\\PAttributesRectsCalculator.hx"
				switch (global::Type.enumIndex(time)){
					case 16:
					{
						#line 42 "F:\\nx3\\src\\nx3\\PAttributesRectsCalculator.hx"
						return new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(((double) (0) ), true), new global::haxe.lang.Null<double>(((double) (-3) ), true), new global::haxe.lang.Null<double>(((double) (6) ), true), new global::haxe.lang.Null<double>(((double) (6) ), true));
					}
					
					
					default:
					{
						#line 45 "F:\\nx3\\src\\nx3\\PAttributesRectsCalculator.hx"
						return new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(((double) (0) ), true), new global::haxe.lang.Null<double>(((double) (-3) ), true), new global::haxe.lang.Null<double>(((double) (4) ), true), new global::haxe.lang.Null<double>(((double) (3) ), true));
					}
					
				}
				
			}
			#line default
		}
		
		
		public static  new object __hx_createEmpty(){
			unchecked {
				#line 8 "F:\\nx3\\src\\nx3\\PAttributesRectsCalculator.hx"
				return new global::nx3.PAttributesRectsCalculator(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
			}
			#line default
		}
		
		
		public static  new object __hx_create(global::Array arr){
			unchecked {
				#line 8 "F:\\nx3\\src\\nx3\\PAttributesRectsCalculator.hx"
				return new global::nx3.PAttributesRectsCalculator();
			}
			#line default
		}
		
		
	}
}

File: Playground/TestDiff/java/src/nx3/PAttributesRectsCalculator.java
package nx3;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class PAttributesRectsCalculator extends haxe.lang.HxObject
{
	public    PAttributesRectsCalculator(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    PAttributesRectsCalculator()
	{
		nx3.PAttributesRectsCalculator.__hx_ctor_nx3_PAttributesRectsCalculator(this);
	}
	
	
	public static   void __hx_ctor_nx3_PAttributesRectsCalculator(nx3.PAttributesRectsCalculator __temp_me67)
	{
		{
		}
		
	}
	
	
	public static   nx3.geom.Rectangle getClefRect(nx3.EClef clef)
	{
		switch (haxe.root.Type.enumIndex(clef))
		{
			case 2:
			{
				return new nx3.geom.Rectangle(((java.lang.Object) (0) ), ((java.lang.Object) (-3) ), ((java.lang.Object) (9) ), ((java.lang.Object) (6) ));
			}
			
			
			case 1:
			{
				return new nx3.geom.Rectangle(((java.lang.Object) (0) ), ((java.lang.Object) (-4) ), ((java.lang.Object) (9) ), ((java.lang.Object) (8) ));
			}
			
			
			case 0:
			{
				return new nx3.geom.Rectangle(((java.lang.Object) (0) ), ((java.lang.Object) (-5) ), ((java.lang.Object) (9) ), ((java.lang.Object) (10) ));
			}
			
			
		}
		
		return null;
	}
	
	
	public static   nx3.geom.Rectangle getKeyRect(nx3.EKey key)
	{
		switch (haxe.root.Type.enumIndex(key))
		{
			case 6:
			{
				return new nx3.geom.Rectangle(((java.lang.Object) (0) ), ((java.lang.Object) (-3) ), ((java.lang.Object) (1) ), ((java.lang.Object) (6) ));
			}
			
			
			case 7:case 5:
			{
				return new nx3.geom.Rectangle(((java.lang.Object) (0) ), ((java.lang.Object) (-3) ), ((java.lang.Object) (4.4) ), ((java.lang.Object) (6) ));
			}
			
			
			case 8:case 4:
			{
				return new nx3.geom.Rectangle(((java.lang.Object) (0) ), ((java.lang.Object) (-3) ), ((java.lang.Object) (6.8) ), ((java.lang.Object) (6) ));
			}
			
			
			case 9:case 3:
			{
				return new nx3.geom.Rectangle(((java.lang.Object) (0) ), ((java.lang.Object) (-3) ), ((java.lang.Object) (9.2) ), ((java.lang.Object) (6) ));
			}
			
			
			case 10:case 2:
			{
				return new nx3.geom.Rectangle(((java.lang.Object) (0) ), ((java.lang.Object) (-3) ), ((java.lang.Object) (11.6) ), ((java.lang.Object) (6) ));
			}
			
			
			case 11:case 1:
			{
				return new nx3.geom.Rectangle(((java.lang.Object) (0) ), ((java.lang.Object) (-3) ), ((java.lang.Object) (14.) ), ((java.lang.Object) (6) ));
			}
			
			
			default:
			{
				return new nx3.geom.Rectangle(((java.lang.Object) (0) ), ((java.lang.Object) (-2) ), ((java.lang.Object) (.5) ), ((java.lang.Object) (4) ));
			}
			
		}
		
	}
	
	
	public static   nx3.geom.Rectangle getTimeRect(nx3.ETime time)
	{
		if (( time == null )) 
		{
			return new nx3.geom.Rectangle(((java.lang.Object) (0) ), ((java.lang.Object) (-3) ), ((java.lang.Object) (.5) ), ((java.lang.Object) (3) ));
		}
		
		switch (haxe.root.Type.enumIndex(time))
		{
			case 16:
			{
				return new nx3.geom.Rectangle(((java.lang.Object) (0) ), ((java.lang.Object) (-3) ), ((java.lang.Object) (6) ), ((java.lang.Object) (6) ));
			}
			
			
			default:
			{
				return new nx3.geom.Rectangle(((java.lang.Object) (0) ), ((java.lang.Object) (-3) ), ((java.lang.Object) (4) ), ((java.lang.Object) (3) ));
			}
			
		}
		
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new nx3.PAttributesRectsCalculator(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new nx3.PAttributesRectsCalculator();
	}
	
	
}


